CONTRIBUTING.md

```diff
@@ -0,0 +1,80 @@
+    - [ ] The developer added test cases to cover integrating with library
+    - [ ] If extending a third-party component, you have validated that is the best design solution
+- HTML/CSS/UI Controls
+  - [ ] HTML markup and CSS classes are semantic and meaningful
+  - [ ] Styles are applied by classes, not through style tags on elements
+  - [ ] CSS class names reflect structure, state, or intent rather than specific styles
+  - [ ] Markup choices reflect intent, e.g., `button` for user actions, `a` for navigation
+  - Is the developer creating a new UI component or extending an existing component?
+    - [ ] There is no existing UI component that can be used or modified
+    - [ ] Keyboard interactions follow the conventions listed in the WAI-ARIA Authoring Practices
+    - [ ] Aria roles and attributes are applied as necessary
+  - Code Smells
+    - [ ] Lots of custom CSS classes require careful review
+    - [ ] Component inheritance requires careful review
+    - [ ] Class names with specific styles require careful review
+    - [ ] Large numbers of divs or nested divs require careful review
+    - [ ] No aria tags on custom controls with many event handlers requires careful review
+- Modularity & Decomposition
+  - Is the PR for a new feature?
+    - [ ] The feature is written as a standalone module
+    - [ ] The feature is lazy loaded unless required during App initialization. This applies to both top level routing and children of feature modules
+    - [ ] Only the code necessary to execute the feature is included in the module
+    - [ ] Features do not expose internal implementation via exports
+    - [ ] Modules consuming shared components explicitly import their dependencies (Do not assume deps are provided by a higher module like App)
+  - Is the PR for a UI or data-access component, e.g., a class in app-components or api-kit?
+    - [ ] The component is declared and exported in its own module
+    - [ ] The component is not included in a "big bang" module (AppComponentsModule, ApiKitModule). Each consuming module must explicitly import all dependencies.
+  - Is the code a utility library?
+    - [ ] Any Angular utilities follow the same module rules as UI and data-access components
+    - [ ] JS/TS POJOs or functions should be exported from their respective files inside an appropriately named folder that includes unit tests, interfaces, or any other code related to that.
+  - [ ] Barrel files (index.js/ts) are avoided (this creates complications with Angular Package Format)
+  - Code Smells
+    - [ ] Classes and templates that exceed ~1-300 LOC require careful review
+    - [ ] Templates that make excessive use of conditionals require careful review
+    - [ ] Modules do not have large numbers of declarations (> 5 requires careful review)
+      - [ ] If more than 5 declarations, ensure that the module cannot be broken down into smaller modules and composed together again with imports
```

Comment:
@thomastighe 

I was hoping it would be clear from the language that does not mean "Not allowed to merge if more than 5 declarations". I want the language to convey that as a reviewer, if you see a module that has a large number of declarations, the reviewer should take an extra careful look at what each thing in the module is doing. 

On our project, we have had modules with dozens of declarations, so saying 5 is a way of level setting that a small module might have 1-3 declarations, a medium sized module might have double that, and a large module is anything bigger than that. 

The goal is to get to the point where the module only contains exactly the items needed to achieve a purpose with everything else left out. A module might actually need 100 components, and if that is the minimum number of modules required to achieve a purpose, that would be fine.

However, its really hard to come up with a heuristic for a reviewer to be able to scan code and catch these kinds of issues. The point of this is that if you see a module with declarations above an arbitrary number, you as the reviewer should spend more time reviewing that code to make sure all the declarations are helping achieve a single purpose.

But that's hard to write in a one liner for a checklist :)

Reply:
I fully agree I just wanted this conversation written out somewhere to say that this is not strict but a guide and something to question if large numbers. Are we really thinking about abstraction levels.